pkg/proxyagent/agent/agent.go

```diff
@@ -126,9 +65,89 @@ func (p *proxyAgent) GetAgentAddonOptions() agent.AgentAddonOptions {
 			CSRApproveCheck: func(cluster *clusterv1.ManagedCluster, addon *addonv1alpha1.ManagedClusterAddOn, csr *csrv1.CertificateSigningRequest) bool {
 				return cluster.Spec.HubAcceptsClient
 			},
-			PermissionConfig: p.setupPermission,
+			PermissionConfig: NewClusterProxySetupPermissionFunc(runtimeClient, nativeClient),
 			CSRSign:          CustomSignerWithExpiry(ProxyAgentSignerName, caKeyData, caCertData, time.Hour*24*180),
-		},
+		}).
+		WithInstallStrategy(agent.InstallAllStrategy(common.AddonInstallNamespace)).
+		WithScheme(scheme).
```

Comment:
we could ignore `WithScheme` if we have no customized resources. anyway it is ok to configure.

Reply:
removed